cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -137,6 +152,8 @@ void Run(const std::vector<string>& bag_filenames) {
     rosbag::View view(bag);
     const ::ros::Time begin_time = view.getBeginTime();
     const double duration_in_seconds = (view.getEndTime() - begin_time).toSec();
+    // The message processing loop publishes the clock.
```

Comment:
No. Please make it more readable. Putting a `start()` call before `SerializeState()` is fine. In general it is a good idea to write your code like you explain it to a human ("readability"). You explain your code as publishing during `FinishTrajectory()`, `SerializeState()` and waiting for shutdown, so unless there is a good reason to deviate from this, please implement it in the same way.